### ISSUE TITLE
Visio diagram not being generated

### ISSUE THREAD

**Issue by spaelling** [microsoft/ARI#149]
the code for that part is commented out. I am looking for the network topology diagram, which I do no see in the draw.io diagram?

**Comment by Giupy75** [comment]
Hi good job , but i'm same issues , i can't find diagram.vsdx.  the powershell generated only  xlsx and xml file .

**Comment by Claudio-Merola** [comment]
The xml file is being generated?

**Comment by spaelling** [comment]
yes, but I see no network topology in that. My understanding is that the network topology is in the Visio diagram?

**Comment by crimez** [comment]
I guess I'm experiencing the same problem, no "Network topology" sheet in the Diagram file. I have to include the parameter -DiagramFullEnvironment in order to get the topology

**Comment by red-erik** [comment]
Hello,
I confirm no network topology with of without "DiagramFullEnvironment" parameter.
Regards,
Red.

**Comment by Claudio-Merola** [comment]
I'm not being able to replicate this issue.  

1) Do you guys have Virtual Network Gateways in the environment?

2) There is Express Route in the environment?

3) There is vWAN in the environment?

**Comment by red-erik** [comment]
Hello Claudio,
yes, we have all of them.
Regards,
Red.

**Comment by Claudio-Merola** [comment]
Hi @red-erik 

Can you try the last version of the script and see if this still happens?

**Comment by red-erik** [comment]
Hello,
to give you more detailed feedback, it works on small Subs with not too many objects while on larger ones, it does not produce any output (Diagram).
Regards,
Red.

**Comment by crimez** [comment]
I'm still having the issue even with the last version of the script

**Comment by red-erik** [comment]
Hello,
just as a reminder, the code inside the script is commented out so you need to uncomment it

<######################################################### VISIO DIAGRAM JOB ######################################################################>
        <#
        Write-Debug ('Checking if Visio Diagram Job Should be Run.')
        if ($Diagram.IsPresent) {
            Write-Debug ('Starting Visio Diagram Processing Job.')
            Start-job -Name 'VisioDiagram' -ScriptBlock {

                If ($($args[5]) -eq $true) {
                    $ModuSeq = (New-Object System.Net.WebClient).DownloadString($($args[7]) + '/Extras/VisioDiagram.ps1')
                }
                Else {
                    $ModuSeq0 = New-Object System.IO.StreamReader($($args[0]) + '\Extras\VisioDiagram.ps1')
                    $ModuSeq = $ModuSeq0.ReadToEnd()
                    $ModuSeq0.Dispose()  
                }                  

                $ScriptBlock = [Scriptblock]::Create($ModuSeq)

                $VisioRun = ([PowerShell]::Create()).AddScript($ScriptBlock).AddArgument($($args[1])).AddArgument($($args[2])).AddArgument($($args[3])).AddArgument($($args[4]))

                $VisioJob = $VisioRun.BeginInvoke()

                while ($VisioJob.IsCompleted -contains $false) {}

                $VisioRun.EndInvoke($VisioJob)

                $VisioRun.Dispose()

            } -ArgumentList $PSScriptRoot, $Subscriptions, $Resources, $Advisories, $DFile, $RunOnline, $Repo, $RawRepo   | Out-Null
        }
        #>

**Comment by Claudio-Merola** [comment]
Ohhh I see.

We are no longer working with VISIO diagram...

At the moment we only develop the Draw.io diagram...

**Comment by red-erik** [comment]
Thank you for clarifying that.

Regards,
Red.
